sdk/core/azure-core/inc/azure/core/internal/cryptography/sha_hash.hpp

```diff
@@ -17,6 +17,62 @@
 
 namespace Azure { namespace Core { namespace Cryptography { namespace _internal {
 
+  /**
+   * @brief Defines #Sha1Hash.
+   *
+   * @warning SHA1 is a deprecated hashing algorithm and SHOULD NOT be used,
+   * unless it is used to implement a specific protocol (for instance, RFC 6455 and
+   * RFC 7517 both require the use of SHA1 hashes). SHA256, SHA384, and SHA512 are all preferred to
+   * SHA1.
+   *
+   */
+  class Sha1Hash final : public Azure::Core::Cryptography::Hash {
```

Comment:
If this is only used by the websocket implementation, I would move all Sha1Hash to .cpp files and keep it private.

Unless the websocket API is expecting a `Sha1` as input, instead of internally generating the Sha1Hash as part of the protocol, I would move this to be implementation details.

Reply:
That would require pulling all the OpenSSL goo into WebSockets implementation. It also means we have two different hashing implementations in core.

Sha1 hashes are used for multiple things that have nothing to do with websockets, including X.509 certificate thumbprints, so having this in core makes a lot of sense.

If it had been there when I was writing attestation, I'd have used it instead of building a private version in attestation.

Reply:
All I'm saying is that this comment:

> @warning SHA1 is a deprecated hashing algorithm and SHOULD NOT be used,

Makes me think if we should be then adding this code to the public.  It is kind of weird to add something and commented as `SHOULD NOT be used....`  .  That's why I would prefer having it at implementation level only, where we are sure that it is OK, and not exposed to the world.  

But, your call,